exporter/sematextexporter/writer.go

```diff
@@ -0,0 +1,241 @@
+		}
+	}
+	queryValues := writeURL.Query()
+
+	writeURL.RawQuery = queryValues.Encode()
+
+	return writeURL.String(), nil
+}
+
+// Start implements component.StartFunc
+func (w *sematextHTTPWriter) Start(ctx context.Context, host component.Host) error {
+	httpClient, err := w.httpClientSettings.ToClient(ctx, host, w.telemetrySettings)
+	if err != nil {
+		return err
+	}
+	w.httpClient = httpClient
+	return nil
+}
+
+func (w *sematextHTTPWriter) NewBatch() otel2influx.InfluxWriterBatch {
+	return newSematextHTTPWriterBatch(w)
+}
+
+var _ otel2influx.InfluxWriterBatch = (*sematextHTTPWriterBatch)(nil)
+
+type sematextHTTPWriterBatch struct {
+	*sematextHTTPWriter
+	encoder      *lineprotocol.Encoder
+	payloadLines int
+}
+
+func newSematextHTTPWriterBatch(w *sematextHTTPWriter) *sematextHTTPWriterBatch {
+	return &sematextHTTPWriterBatch{
+		sematextHTTPWriter: w,
+	}
+}
+
+// EnqueuePoint emits a set of line protocol attributes (metrics, tags, fields, timestamp)
+// to the internal line protocol buffer.
+// If the buffer is full, it will be flushed by calling WriteBatch.
+func (b *sematextHTTPWriterBatch) EnqueuePoint(ctx context.Context, measurement string, tags map[string]string, fields map[string]any, ts time.Time, _ common.InfluxMetricValueType) error {
+	if b.encoder == nil {
+		b.encoder = b.encoderPool.Get().(*lineprotocol.Encoder)
+	}
+	// Add token and os.host tags
+	if tags == nil {
+		tags = make(map[string]string)
+	}
+	tags["token"] = b.token
+	tags["os.host"] = b.hostname
+
+	b.encoder.StartLine(measurement)
+	for _, tag := range b.optimizeTags(tags) {
+		b.encoder.AddTag(tag.k, tag.v)
+	}
+	for k, v := range b.convertFields(fields) {
+		b.encoder.AddField(k, v)
+	}
+	b.encoder.EndLine(ts)
+
+	if err := b.encoder.Err(); err != nil {
+		b.encoder.Reset()
+		b.encoder.ClearErr()
+		b.encoderPool.Put(b.encoder)
+		b.encoder = nil
+		return consumererror.NewPermanent(fmt.Errorf("failed to encode point: %w", err))
+	}
+
+	b.payloadLines++
+	if b.payloadLines >= b.payloadMaxLines || len(b.encoder.Bytes()) >= b.payloadMaxBytes {
+		if err := b.WriteBatch(ctx); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// WriteBatch sends the internal line protocol buffer to Sematext.
+func (b *sematextHTTPWriterBatch) WriteBatch(ctx context.Context) error {
+	if b.encoder == nil {
+		fmt.Println("Encoder is nil")
+		return nil
+	}
+
+	defer func() {
+		b.encoder.Reset()
+		b.encoder.ClearErr()
+		b.encoderPool.Put(b.encoder)
+		b.encoder = nil
+		b.payloadLines = 0
+	}()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, b.writeURL, bytes.NewReader(b.encoder.Bytes()))
+	if err != nil {
+		return consumererror.NewPermanent(err)
+	}
+	res, err := b.httpClient.Do(req)
+	if err != nil {
+		fmt.Println("Error sending request:", err)
+		return err
+	}
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return err
+	}
+
+	if err = res.Body.Close(); err != nil {
+		return err
+	}
+
+	switch res.StatusCode / 100 {
+	case 2:
```

Comment:
any reason for the extra math here? Might be clearer to just go with `case 200`, `case 500`, etc.

Reply:
corrected